src/ffi/spatialmath/vector3.rs

```diff
@@ -0,0 +1,121 @@
+use ffi_helpers::null_pointer_check;
+use libc::c_double;
+
+use crate::spatialmath::vector3::Vector3;
+
+#[no_mangle]
+pub extern "C" fn new_vector3(x: f64, y: f64, z: f64) -> *mut Vector3 {
+    Vector3::new(x, y, z).to_raw_pointer()
+}
+
+#[no_mangle]
+pub extern "C" fn free_vector_memory(ptr: *mut Vector3) {
+    if ptr.is_null() {
+        return;
+    }
+    unsafe {
```

Comment:
Why are we not putting the `unsafe` in the signature instead? seems more consistent with how we do this everywhere else.

Reply:
yeah my bad